Formula/mercurial.rb

```diff
@@ -21,17 +22,20 @@ class Mercurial < Formula
     sha256 x86_64_linux:   "3e2f28ff2702e50f54199485a43cb3719593ea462ff0ffc0365fd87c02bac407"
   end
 
-  depends_on "python@3.9"
+  depends_on "python@3.10"
 
   def install
     ENV["HGPYTHON3"] = "1"
 
-    system "make", "PREFIX=#{prefix}", "PYTHON=python3", "install-bin"
+    system "make", "PREFIX=#{prefix}",
+                   "PYTHON=#{Formula["python@3.10"].opt_bin/"python3"}",
```

Comment:
```suggestion
                   "PYTHON=#{which("python3")}",
```
Using `python3` should be fine, though, since Homebrew Python 3.10 should be the first `python3` in `PATH` during the build.

Reply:
In `mercurial`'s Makefile, they run `$PYTHON setup.py install` to install Python packages. Since we change the shebangs of `hg-fast-export.py` in the formula `hg-fast-export`, I think it's better to force the exact path to the versioned Python executable.

Reply:
Doesn't `$PYTHON setup.py install` create a shebang using the path to `$PYTHON` even if `$PYTHON` is not an absolute path?

Reply:
Yeh, change to `which("python3")` would work fine.

> since Homebrew Python 3.10 should be the first `python3` in `PATH` during the build.

What if the first `python3` in `PATH` is the system Python rather than a brewed one since `python@3.10` is still keg-only.

Reply:
> What if the first python3 in PATH is the system Python rather than a brewed one since python@3.10 is still keg-only.

It's not. Keg-only formulae are in `PATH` during `install`.

Reply:
> It's not. Keg-only formulae are in PATH during install.

The CI tests of `hg-fast-export` indicate that it cannot find a `python3` installed with `mercurial` packages.

1. install `mercurial`:

    The `mercurial` installer runs `$PYTHON setup.py install`.

2. install `hg-fast-export`.

3. test `hg-fast-export`:

    in `hg-fast-export.sh`, it runs:

    ```bash
    for python_cmd in python2 python python3; do
        # test $python_cmd is executable
        # test $python_cmd installed with the `mercurial` packages installed in step 1.
    done
    ```

If the `python3` executable in PATH for `mercurial` and `hg-fast-export` are the same ( both keg-only `python3.10`). Why `hg-fast-export.sh` could not find a python executable installed with needed Python packages in step 3. above.

Reply:
> Why `hg-fast-export.sh` could not find a Python executable installed with needed Python packages in step 3. above

Because step 3 runs outside of `install`, so keg-only formulae are not in `PATH`.

Reply:
So `which("python3") == Formula["python@3.10"].opt_bin/"python3"` in `install`?

Reply:
Yes. And calling `python3` in `install` will find `python3.10`.